standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/metrics/Metrics.java

```diff
@@ -0,0 +1,210 @@
+    registry = new MetricRegistry();
+
+    registry.registerAll(new GarbageCollectorMetricSet());
+    registry.registerAll(new BufferPoolMetricSet(ManagementFactory.getPlatformMBeanServer()));
+    registry.registerAll(new MemoryUsageGaugeSet());
+    registry.registerAll(new ThreadStatesGaugeSet());
+    registry.registerAll(new ClassLoadingGaugeSet());
+
+    /*
+     * This is little complicated.  First we look for our own config values on this.  If those
+     * aren't set we use the Hive ones.  But Hive also has multiple ways to do this, so we need to
+     * look in both of theirs as well.  We can't use theirs directly because they wrap the
+     * codahale reporters in their own and we do not.
+     */
+    // Check our config value first.  I'm explicitly avoiding getting the default value for now,
+    // as I don't want our default to override a Hive set value.
+    String reportersToStart = conf.get(MetastoreConf.ConfVars.METRICS_REPORTERS.varname);
+    if (reportersToStart == null) {
+      // Now look in the current Hive config value.  Again, avoiding getting defaults
+      reportersToStart =
+          conf.get(MetastoreConf.ConfVars.HIVE_CODAHALE_METRICS_REPORTER_CLASSES.hiveName);
+      if (reportersToStart == null) {
+        // Last chance, look in the old Hive config value.  Still avoiding defaults.
+        reportersToStart =
+            conf.get(MetastoreConf.ConfVars.HIVE_METRICS_REPORTER.hiveName);
+        if (reportersToStart == null) {
+          // Alright fine, we'll use our defaults
+          reportersToStart = MetastoreConf.getVar(conf, MetastoreConf.ConfVars.METRICS_REPORTERS);
+        }
+      }
+    }
+
+    reporters = new ArrayList<>();
+    scheduledReporters = new ArrayList<>();
+    if (reportersToStart != null && reportersToStart.length() > 0) {
+      String[] reporterNames = reportersToStart.toLowerCase().split(",");
+      for (String reporterName : reporterNames) {
+        if (reporterName.equals("console") || reporterName.endsWith("consolemetricsreporter")) {
+          ConsoleReporter reporter = ConsoleReporter.forRegistry(registry)
+              .convertRatesTo(TimeUnit.SECONDS)
+              .convertDurationsTo(TimeUnit.MILLISECONDS)
+              .build();
+          reporter.start(15, TimeUnit.SECONDS);
+          reporters.add(reporter);
+          scheduledReporters.add(reporter);
+        } else if (reporterName.equals("jmx") || reporterName.endsWith("jmxmetricsreporter")) {
+          JmxReporter reporter = JmxReporter.forRegistry(registry)
+              .convertRatesTo(TimeUnit.SECONDS)
+              .convertDurationsTo(TimeUnit.MILLISECONDS)
+              .build();
+          reporter.start();
+          reporters.add(reporter);
+        } else if (reporterName.startsWith("json") || reporterName.endsWith("jsonfilemetricsreporter")) {
+          // We have to initialize the thread pool before we start this one, as it uses it
+          ThreadPool.initialize(conf);
```

Comment:
Curious to understand how JsonReporter uses the ThreadPool. I don't see the code which explicitly makes use of this class. Would be good to add another line here describing how it uses it? Sorry for pestering review comments but I think would help in the longer run.

Reply:
It looks like that was a holdover.  JsonFileMetricsReporter (which I based JsonReporter after) does use the ThreadPool, but JsonReporter does not.  I'll remove the initialization before I commit.  I'm not going to remove ThreadPool from the patch though, as I'll need it later anyway and since I've already put there I might as well leave it.

Reply:
sounds good. Thanks!